packages/core/content-manager/server/src/history/services/history.ts

```diff
@@ -422,16 +206,16 @@ const createHistoryService = ({ strapi }: { strapi: Core.Strapi }) => {
         FIELDS_TO_IGNORE,
         contentTypeSchemaAttributes
       ) as Struct.SchemaAttributes;
+
       // Set all deleted relation values to null
-      const dataWithoutMissingRelations = await Object.entries(sanitizedSchemaAttributes).reduce(
+      const reducer = async.reduce(Object.entries(sanitizedSchemaAttributes));
+      const dataWithoutMissingRelations = await reducer(
         async (
-          previousRelationAttributesPromise: Promise<Record<string, unknown>>,
+          previousRelationAttributes: Record<string, unknown>,
           [name, attribute]: [string, Schema.Attribute.AnyAttribute]
         ) => {
```

Comment:
Sicne the function is async, I don't understand how the accumulator value can not be a promise 🤔

Reply:
I believe it is resolved [here](https://github.com/strapi/strapi/blob/5c8c735ac09d862fe8bd7a517d02b76ce4d7dfbc/packages/core/utils/src/async.ts#L34-L35) inside the async reducer

Reply:
ah i thought it was the classic Array.reduce